catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -207,13 +215,32 @@ private void fetchNextResults() {
       try {
         SourceResponse response = queryFunction.query(queryRequestCopy);
 
-        if (response.getResults().size() == 0) {
+        final List<Result> resultList = response.getResults();
+        int resultCount = resultList.size();
+
+        final List<Result> dedupedResults =
+            resultList.stream().filter(this::isDistinctResult).collect(Collectors.toList());
+
+        // TODO RAP 19 Sep 17: This test is insufficient, as it does not account
+        // for large blocks of filtered results
+        if (resultCount == 0) {
           finished = true;
           return;
         }
 
-        results = response.getResults().iterator();
-        currentIndex += response.getResults().size();
+        dedupedResults
```

Comment:
❓ Since we have to assume that results will always be returned in the same order for paging to work properly, I was thinking that we might be able to only keep the ID of the _last_ metacard returned and drop everything that comes prior and including that metacard. Thoughts?

Reply:
It is possible to get duplicate results because the catalog changes between successive queries. Is it also possible for the provider's index to be out-of-sync with the expected results?

Reply:
The index could absolutely be out-of-sync. Have discussed with @lessarderic and we're going to stick with the more expensive but surer deduping logic that's already present.